shakenfist/external_api/app.py

```diff
@@ -677,22 +694,24 @@ def post(self, name=None, cpus=None, memory=None, network=None, disk=None,
                 if not n:
                     m = 'missing network %s during IP allocation phase' % (
                         netdesc['network_uuid'])
-                    db.enqueue_instance_error(instance.uuid, m)
+                    db.enqueue_instance_error(
+                        instance.static_values['uuid'], m)
                     return error(
                         404, 'network %s not found' % netdesc['network_uuid'])
 
                 with db.get_lock('ipmanager', None,  netdesc['network_uuid'],
                                  ttl=120, op='Network allocate IP'):
                     db.add_event('network', netdesc['network_uuid'], 'allocate address',
-                                 None, None, instance.uuid)
+                                 None, None, instance.static_values['uuid'])
                     ipm = IPManager.from_db(netdesc['network_uuid'])
                     if 'address' not in netdesc or not netdesc['address']:
                         netdesc['address'] = ipm.get_random_free_address()
                     else:
                         if not ipm.reserve(netdesc['address']):
                             m = 'failed to reserve an IP on network %s' % (
                                 netdesc['network_uuid'])
-                            db.enqueue_instance_error(instance.uuid, m)
+                            db.enqueue_instance_error(
```

Comment:
I feel the code churn could be a lot less by moving the change into the class property function?